component/src/main/java/com/celements/photo/utilities/ImageUrlExtractor.java

```diff
@@ -0,0 +1,56 @@
+package com.celements.photo.utilities;
+
+import java.util.List;
+
+import javax.validation.constraints.NotNull;
+
+import org.xwiki.component.annotation.ComponentRole;
+
+import com.celements.photo.container.ImageUrl;
+import com.google.common.base.Optional;
+
+@ComponentRole
+public interface ImageUrlExtractor {
+
+  /**
+   * Returns all relative (internal) image URLs extracted from an HTML snippet.
+   *
+   * @param content
+   *          An HTML snippet.
+   * @return list containing all the images in content
+   */
+  public @NotNull List<ImageUrl> extractImageUrlList(@NotNull String content);
+
+  /**
+   * Returns all relative (internal) image URLs that match certain size criteria to be shown on
+   * social media sites. The list is sorted by image area size ascending.
+   *
+   * @param content
+   *          An HTML snippet.
+   * @return list containing all social media usable images
+   */
+  public @NotNull List<ImageUrl> extractImagesSocialMediaUrlList(@NotNull String content);
```

Comment:
I would argue that this is too specific for the ImageUrlExtractor. the client can now easily do it with Predicates/Comparators, so better move it there (in celements-blog I presume).

perhaps add information about Predicates/Comparators to ImageUrlExtractor javadoc.